library/testing/TestSDKOptionsIO.cxx

```diff
@@ -71,5 +72,16 @@ int TestSDKOptionsIO(int argc, char* argv[])
   test.parse<std::vector<std::string>>(
     "std::vector<std::string>", "  foo, bar ,   baz ", { "foo", "bar", "baz" });
 
+  test.parse<f3d::vector3_t>("vector3_t", "1, 2, 3", { 1, 2, 3 });
+  test.parse<f3d::vector3_t>("vector3_t", " 1,    2, 3  ", { 1, 2, 3 });
+  test.parse<f3d::vector3_t>("vector3_t", "+Y", { 0, 1, 0 });
+  test.parse<f3d::vector3_t>("vector3_t", " +Y ", { 0, 1, 0 });
+  test.parse<f3d::vector3_t>("vector3_t", "-Y", { 0, -1, 0 });
+  test.parse<f3d::vector3_t>("vector3_t", "+X", { 1, 0, 0 });
+  test.parse<f3d::vector3_t>("vector3_t", "-X", { -1, 0, 0 });
+  test.parse<f3d::vector3_t>("vector3_t", "+Z", { 0, 0, 1 });
+  test.parse<f3d::vector3_t>("vector3_t", "-Z", { 0, 0, -1 });
+  test.parse<f3d::vector3_t>("vector3_t", "1, 2", f3d::vector3_t::fromSphericalCoordinates(1, 2));
```

Comment:
Maybe test against cartesian to check spherical to cartesian conversion at the same time:

```suggestion
  test.parse<f3d::vector3_t>("vector3_t", "0, 0", { 0, 0, 1 });
```